### PR TITLE
[DNM] Lima II: Multi-Z Maintenance Shenanigans Turbo Championship Edition

### PR DESCRIPTION
**Pull request by sqnztb**
~~nice~~

Just merge it, what's the worst that can happen?

image

... heh.

Well everyone. Here she is. Lima II: Multi-Z Maintenance Shenanigans Turbo Championship Edition (L2MZMSTCE for short.)

Let's go over the big changes.

**Medical**
- Medical gets some patient recovery rooms. These are semi-private. Close the blinds and the patient cannot be seen by passersby in the hall.
- Morgue shifts down a bit to compensate.
- Viro gets the Z.
- Chemistry gets the Z.

Overview: 
image
Lower Viro:
image
Lower Chem:
image

**Science**
- Xenobio gets the Z
- New lower-z testing lab.

Overview, note the stairs at the top of the image.
image
Lower Xenobio:
image
Lower testing lab:
image

**Service**
Nearly everything in service got touched in some ways. Y'all got a nice pretty service hall now though.
image
Botany bee dungeon.
image

**Engineering/Atmospherics**
HFR goes to lower level. Some of the engineering equipment goes down to the lower level. New breakroom.
image

**Maintenance Changes**
Where applicable, I have changed some of the location-named maintenance area types to department-named maintenance types. E.g. "Starboard Maintenance" to "Science Maints", "Aft Maints" to "Cargo Maints", etc.

And finally, here's the map of the lower Z.
image

Feedback is requested prior to merging this. It's not 100% ready in its current state, but it's close enough to start the pull and get the community feedback. I still have a little bit of detailing to do.

### CODE REVIEW

**Comment by Jolly-66** [comment]
You know there are sub typed department maints right? Check Pubby for ref.

**Comment by sqnztb** [comment]
Yes. I changed several maint areas to them.
> Maintenance Changes
Where applicable, I have changed some of the location-named maintenance area types to department-named maintenance types. E.g. "Starboard Maintenance" to "Science Maints", "Aft Maints" to "Cargo Maints", etc.

image

**Comment by Jolly-66** [comment]
> Yes. I changed several maint areas to them.
> 
> > Maintenance Changes
> > Where applicable, I have changed some of the location-named maintenance area types to department-named maintenance types. E.g. "Starboard Maintenance" to "Science Maints", "Aft Maints" to "Cargo Maints", etc.
> 
> image

Yeah I was half asleep reading the pr my bad

**Comment by MrMelbert** [comment]
image
image

**Comment by MrMelbert** [comment]
image

**Comment by sqnztb** [comment]
Yeah I was working down the list for the test station z issues, was getting into the RC section. Everything else has been fixed.

**Comment by sqnztb** [comment]
Lots of little cleanup. Decals that were hidden underneath wall turfs. Air alarms, newscasters, status displays. Science breakroom. Double checked all of the piping and wiring. Whiteship dock is now properly big enough to fit the frigate (delta) and salvage ship (meta) whiteships, though I don't give two shits about the 13x41 monstrosity of tram's whiteship, fuck that noise.

Some slight non-modularity with the nav computer tweak, though I don't expect this to run into any merge conflicts from tg any time soon. Still researching this so I'm leaving it up as DNM. 

It's ready though. I hope.

**Comment by Jolly-66** [comment]
Did you doc the modular changes?

File: code/modules/shuttle/navigation_computer.dm [hunk 1]
```diff
@@ -20,7 +20,7 @@
 	var/view_range = 0
 	var/x_offset = 0
 	var/y_offset = 0
-	var/list/whitelist_turfs = list(/turf/open/space, /turf/open/floor/plating, /turf/open/lava)
+	var/list/whitelist_turfs = list(/turf/open/space, /turf/open/floor/plating, /turf/open/lava, /turf/open/openspace)
```
Comment (Jolly-66) [inline code/modules/shuttle/navigation_computer.dm:23]: I'm going to kill you

Add this to the Jollystation readme

Comment (sqnztb) [reply code/modules/shuttle/navigation_computer.dm:23]: >  Still researching this so I'm leaving it up as DNM.

Also this got discussed separately in coderbus, I've submitted a pull to tg to fix this issue. I'll revert that file on this when it merges.

**Comment by MrMelbert** [comment]
Science

This room's very plain, not sure if anything else can be done here but eh
image


Plating
image
Fire door
image
One sliver of intercom dead zone, not too important
image
Fire door
image

**Comment by MrMelbert** [comment]
Engineering

Seems like none of these doors have firedoors at all
image

**Comment by MrMelbert** [comment]
Service

Same lack of firedoors around the service hall
image
This far end has no intercom coverage, unsure if intentional
image

**Comment by MrMelbert** [comment]
Medbay 

This disposals outlet leads riiight into a grille
image

**Comment by MrMelbert** [comment]
Other misc notes:

- No carp spawn on the lower levels?
- (Major) Space dust is totally going to make the lower maintenance uninhabitable very often
- The entire lower area being the same area might very issues with area based things, it might be wise to break it up into sections, maybe define some new maintenance areas in the code if you have to.
- Some of the minor fixes and adjustments made to Lima prior didn't make it over to this version, but i'll just port them over myself when this is done and dusted

overall, very gud

**Review by MrMelbert — CHANGES_REQUESTED** [review]
the jannie's back door is a space area
image

**Review by Jolly-66 — CHANGES_REQUESTED** [review]
Add a trap door to the HoP line.

Trust me this'll be funny.

**Comment by sqnztb** [comment]
I'm not going to let you errantly space people from the line.

**Comment by Jolly-66** [comment]
> I'm not going to let you errantly space people from the line.

Just dump them in maint like the rats they are damnit!!!!!

**Comment by Jolly-66** [comment]
Oh yeah don't forget to add the closes thing to close that one rogue lima bug report

**Comment by sqnztb** [comment]
Cleaned up all of the windoors across the station. No more var/iconstate edited garbage like...
```/obj/machinery/door/window/northleft{
    base_state = "right";
    dir = 2;
    icon_state = "right";
    name = "Containment Pen #4";
    req_access_txt = "55"
    },```

Also fixed all of the lower z's grille armor. I'd copied the wrong turf (one that was just grille+openspace instead of grille+rwall) so now space dust shouldn't be anywhere near as bad. Nor should you be able to fall down to central command.

And thank you for reminding me.
Fixes #63

**Comment by Jolly-66** [comment]
Pending upstream merge; some fixes and updates will come after that

@MrMelbert

**Pull request merged**